administrator/models/modellist.php

```diff
@@ -176,16 +176,6 @@ protected function _getListCount($query)
 		return $db->getNumRows();
 	}
 
-	protected function canDelete($record)

```

Comment:
Why you deleted this functions? 
It is more practical and no code duplication

Reply:
and I don't like it when there is a lot of nesting, the code becomes unreadable and incomprehensible. I will give an example of how I would do it, if you like it, you can take a note.
```
   public function checkin($pks = array())
    {
        $pks     = (array) $pks;
        $table   = $this->getTable('Comment', 'JCommentsTable');
        $checkin = property_exists($table, 'checked_out');
        $count   = 0;

        if($checkin === false) {
            return $count;
        }

        foreach ($pks as $pk)
        {
            if (!$table->load($pk) || !$table->checkin($pk)) {
                $this->setError($table->getError());

                return false;
            }
            
            if ($table->checked_out > 0)
            {
                $count++;
            }
        }
        

        return $count;
    }
```

Reply:
> 
> 
> Why you deleted this functions? It is more practical and no code duplication

This code used only in two models. And in these two models only in one place it's used. This is no code dupes.

Reply:
> 
> 
> and I don't like it when there is a lot of nesting, the code becomes unreadable and incomprehensible. I will give an example of how I would do it, if you like it, you can take a note.
> 
> ```
>    public function checkin($pks = array())
>     {
>         $pks     = (array) $pks;
>         $table   = $this->getTable('Comment', 'JCommentsTable');
>         $checkin = property_exists($table, 'checked_out');
>         $count   = 0;
> 
>         if($checkin === false) {
>             return $count;
>         }
> 
>         foreach ($pks as $pk)
>         {
>             if (!$table->load($pk) || !$table->checkin($pk)) {
>                 $this->setError($table->getError());
> 
>                 return false;
>             }
>             
>             if ($table->checked_out > 0)
>             {
>                 $count++;
>             }
>         }
>         
> 
>         return $count;
>     }
> ```

The code for checking is not mine. I just apply a code style. In next update I will use you code and replace. Tanks for suggestion.